components/layout_thread/lib.rs

```diff
@@ -1738,7 +1746,7 @@ impl LayoutThread {
                 ua_or_user: &ua_or_user_guard,
             };
             let snapshots = SnapshotMap::new();
-            let mut layout_context = self.build_layout_context(guards, false, &snapshots);
+            let mut layout_context = self.build_layout_context(guards, false, &snapshots, None);
```

Comment:
Here you might be able to receive the origin as part of the `LayoutControlMsg::TickAnimations`. 

It is sent from https://github.com/servo/servo/blob/a5b43b7df10cc73d8203f7ee0c3a5e1a1bb1d9f5/components/constellation/constellation.rs#L3231

So there you have the `Pipeline` available, so you could to change the below:

https://github.com/servo/servo/blob/a5b43b7df10cc73d8203f7ee0c3a5e1a1bb1d9f5/components/constellation/constellation.rs#L3233

into something like 

```rust
Some(pipeline) => {
    let msg = LayoutControlMsg::TickAnimations(pipeline.load_data.url.origin());
    pipeline.layout_chan.send(msg);
}
```

Reply:
Oh...I was misunderstanding the purpose and use of LayoutContext. And it was a really stupid one :man_facepalming:. And, that `None` was a part of it. I'll correct that now.

Just out of curiosity what is the difference between layout_thread and layout_thread_2020:thinking:?

Reply:
Note that I'm not quite sure about the purpose of `LayoutContext`, I just the `None` there and was thinking that you might not need the option.

`layout_thread_2020` is the "new version" of layout, so your fix might require to be implemented in both, or maybe on 2020? I don't know, @SimonSapin can probably tell you more.

Reply:
Yes, there’s also `components/layout_thread_2020/lib.rs` which is very similar to `components/layout_thread/lib.rs`. See https://github.com/servo/servo/wiki/Layout-2020#using for build instructions.

Reply:
Okay. Earlier, I was under the impression that there might be cases where LayoutContext had to be assigned a new origin and was considering the above code to be a part of that.